### PR TITLE
fix(core): update datepicker while handling via form

### PR DESCRIPTION
**Pull request by Jeff1899**

### CODE REVIEW

File: core/src/components/cat-datepicker/cat-datepicker.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { ErrorMap } from '../cat-form-hint/cat-form-hint';
 import { catI18nRegistry as i18n } from '../cat-i18n/cat-i18n-registry';
 import { getFormat } from './cat-datepicker.config';
 import { getLocale } from './cat-datepicker.locale';
+import { Watch } from '../../../dist/types/stencil-public-runtime';
```
Comment (fynnfeldpausch) [inline core/src/components/cat-datepicker/cat-datepicker.tsx:8]: This seems to be the wrong import path

File: core/src/components/cat-datepicker/cat-datepicker.tsx [hunk 2]
```diff
@@ -249,9 +264,8 @@ export class CatDatepickerFlat {
         onCatChange={e => {
           e.stopPropagation();
           this.pickr?.setDate(e.detail);
```
Comment (fynnfeldpausch) [inline core/src/components/cat-datepicker/cat-datepicker.tsx:266]: if we are setting the value below and the watch kicks in, shouldnt this line be redundant and be deleted?

File: core/src/components/cat-datepicker/cat-datepicker.tsx [hunk 3]
```diff
@@ -249,9 +264,8 @@ export class CatDatepickerFlat {
...
-          if (this.value !== (e.detail || undefined)) {
+          if (this.value !== (e.detail || undefined || '')) {
```
Comment (fynnfeldpausch) [inline core/src/components/cat-datepicker/cat-datepicker.tsx:267]: why do we need || '' here?

**Review by fynnfeldpausch — CHANGES_REQUESTED** [review]

**Review by fynnfeldpausch — APPROVED** [review]

**Comment by hjalmararnorssonhaiilo** [comment]
Hi, I had a look at this fix in the latest Catalyst version, 6.2.3. It seems that this issue is still there as programmatic changes to the bound formcontrol value does not update the selected date in the datepicker UI

**Pull request merged**